sympy/matrices/expressions/tests/test_matexpr.py

```diff
@@ -404,6 +404,18 @@ def test_issue_21195():
     assert A.diff(x) == B
 
 
+def test_issue_24859():
+    A = MatrixSymbol('A', 2, 3)
+    B = MatrixSymbol('B', 3, 2)
+    J = A*B
+    Jinv = Matrix(J).adjugate()
+    u = MatrixSymbol('u', 2, 3)
+    Jk = Jinv.subs(A, A + x*u)
+
+    assert Jk[0,0].diff(x).simplify() == B[0, 1]*u[1, 0] + B[1, 1]*u[1, 1] + B[2, 1]*u[1, 2]
+    assert diff(Jk[0,0], x).doit().simplify() == B[0, 1]*u[1, 0] + B[1, 1]*u[1, 1] + B[2, 1]*u[1, 2]
```

Comment:
Three things:

1. These lines are quite long. Can you refactor the RHSs of the equality checks to an assigned variable that can be reused, something like `expected = B[0, 1]*u[1, 0] + B[1, 1]*u[1, 1] + B[2, 1]*u[1, 2]`.
2. Can we please format the matrix indexing consistently, i.e. change the `Jk[0,0]` to `Jk[0, 0]`.
3. Calls to `simplify` should be avoided if possible because it uses a heuristic approach. If you know the type of simplification needed to achieve the desired result then use one of the [targeted simplification functions](https://docs.sympy.org/latest/tutorials/intro-tutorial/simplification.html). Running locally it looks like the `simplify` isn't needed.

```suggestion
    expected = B[0, 1]*u[1, 0] + B[1, 1]*u[1, 1] + B[2, 1]*u[1, 2]
    assert Jk[0, 0].diff(x) == expected
    assert diff(Jk[0, 0], x).doit() == expected
```